bundles/framework/announcements/instance.js

```diff
@@ -18,8 +18,47 @@ Oskari.clazz.define('Oskari.framework.bundle.announcements.AnnouncementsBundleIn
     }, {
 
         afterStart: function () {
-        }
+            var me = this;
+            if (me.started) {
+                return;
+            }
+            me.started = true;
+            me.sandbox = Oskari.getSandbox();
+            me.sandbox.register(me);
+
+            me.announcementsService = Oskari.clazz.create('Oskari.framework.announcements.service.AnnouncementsService', me.sandbox);
+            me.sandbox.registerService(me.announcementsService);
+
+            if (me.conf && me.conf.plugin) {
+                const mapModule = me.sandbox.findRegisteredModuleInstance('MainMapModule');
+                const plugin = Oskari.clazz.create('Oskari.framework.announcements.plugin.AnnouncementsPlugin', me.conf.plugin.config);
+                mapModule.registerPlugin(plugin);
+                mapModule.startPlugin(plugin);
+            }
 
+            // RPC function to get announcements
+            Oskari.on('app.start', function (details) {
+                const rpcService = Oskari.getSandbox().getService('Oskari.mapframework.bundle.rpc.service.RpcService');
+                if (!rpcService) {
+                    return;
+                }
+                rpcService.addFunction('getAnnouncements', function () {
+                    return new Promise((resolve) => {
+                        me.announcementsService.fetchAnnouncements(announcements => resolve(announcements));
+                    });
+                });
+            });
+        },
+
+        /**
+         * @method update
+         * implements BundleInstance protocol update method - does nothing atm
+         */
+        stop: function () {
+            this.sandbox = null;
+            jQuery('div#announcements-popup').remove();
```

Comment:
Is the popup.remove() needed here? What popup is it? Could you save a reference to it and use the ref to close it instead of removing an element with that id from the DOM?

Reply:
No, seem to have missed few things while doing the last cleanups!